internal/pkg/config/testdata/src/configv2/discussion/test-config-v2.yaml

```diff
@@ -0,0 +1,66 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+source:
+  value:
+    - id: "Tokens are secret"
+      package: configv2/discussion
+      type: Token
+    - id: "TokenBundles are entirely secret"
+      package: configv2/discussion
+      type: TokenBundle
+    - id: "NamedToken.token is secret, NamedToken.name is not"
+      package: configv2/discussion
+      type: NamedToken
+      fieldName: token
+    - id: "Tagged fields are secret, neighboring untagged fields are not"
+      # empty package and type fields match anything
+      fieldTag:
+      - myTag: source
+    - id: "In context, the parameter of SaveSecret(secret interface{}) is a source"
+      # this could use additional details and perhaps a better name than "scope"
+      scope: parameter
+      context:
+        package: configv2/discussion
+        functionName: SaveSecret
+  call:
+    - id: "sensitive environment information"
+      package: os
+      function: getEnv
+      arguments:
+      - position: 0
+        const: CFSSL_CA_PK_PASSWORD
```

Comment:
This feels rather brittle. What if the position changes? (Not applicable in this case, but let's assume the function has more than 1 parameter.) What if it's a variable instead of a constant (same value)?

Reply:
(In both of those cases, hopefully the analysis validation would produce a report saying that this no longer matches.)

But I'm legitimately worried about the case where an argument is a variable. What are the other possibilities for what "arguments" could be?

Reply:
> This feels rather brittle. What if the position changes? (Not applicable in this case, but let's assume the function has more than 1 parameter.) What if it's a variable instead of a constant (same value)?

I don't disagree on any particular point.  Again, I've mostly been trying to reproduce #89 and the associated discussion.  I agree it's brittle to refactoring and anything other that constant arguments.  But this particular case is [TOB-K8S-005](https://github.com/kubernetes/kubernetes/issues/81117), and I would love to defend against such things.

> But I'm legitimately worried about the case where an argument is a variable. What are the other possibilities for what "arguments" could be?

Here's is a gnarly edge-case I've run into, though.  Supposing my sinks are log calls, if I have a sanctified log wrapper `SafeLogf` that cleans all its arguments, I wouldn't want to report any issue if I pass sources to that.  But I might want to report an issue if the format string itself is tainted.

```golang
func Foo(s Source) {
  tainted := fmt.Sprintf("something something %v blah blah", s.Data)
  SafeLogf(tainted + " and something else: %v", s)  // The s will be sanitized, but the format string has the source data hardcoded into it.
}
```

I think the most compelling \*ahem\* argument for argument-level specification is being able to specify which arguments in a sink get sunk, which arguments in a sanitizer get sanitized, etc.  The above example, for instance, could be configured as a sink that only sinks the first argument.  I only have a vague idea about how I might approach that as an implementation, and I agree that maybe this initial draft doesn't cover well how that configuration might look.  But I was trying to leave room to be extensible in that direction.

A lot of these `NotImplemented` portions fit that bill.  It's granularity that isn't an immediate goal, but I'm trying to accommodate extensibility in that direction.  Since they're essentially dead code, I would be happy to prune much of the `NotImplemented` portions before this PR (or another superseding it) were to be merged.  This PR was mostly about fleshing out these discussions.

Reply:
> I don't disagree on any particular point. Again, I've mostly been trying to reproduce #89 and the associated discussion. I agree it's brittle to refactoring and anything other that constant arguments. But this particular case is TOB-K8S-005, and I would love to defend against such things.

Agreed. (Weirdly, I don't recall ever seeing the string "TOB-K8S-005" before. To me it's just been "finding 9 in the k8s security audit". I'll remember it going forward, though.)

> I think the most compelling *ahem* argument for argument-level specification is being able to specify which arguments in a sink get sunk, which arguments in a sanitizer get sanitized, etc.

I agree, that does sound valuable.

I think it would be straightforward to implement since in the `ssa` it's easy to find which argument is in which position in a `Call`.

Going back to the TOB-K8S-005 example, I think one thing that could give us more robustness would be to track variable values, but I think that's only possible under an interpretation approach, and there are certainly cases where it would break down.